components/VolunteerDropdown.tsx

```diff
@@ -0,0 +1,199 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { View, StyleSheet, Platform, Text, TextInput } from 'react-native';
+import {
+  AutocompleteDropdown,
+  TAutocompleteDropdownItem,
+  AutocompleteDropdownRef,
+} from 'react-native-autocomplete-dropdown';
+import { EdgeInsets, useSafeAreaInsets } from 'react-native-safe-area-context';
+import {
+  dropdownDebounceTime,
+  dropdownInputHeight,
+  dropdownListMaxHeight,
+} from '../constants/layout';
+import { useNucaTheme as useTheme } from '../hooks/useNucaTheme';
+import { User } from '../models/User';
+import { loadUsersRequest } from '../utils/users';
+
+type VolunteerDropdownProps = {
+  volunteer?: User;
+  onSelect: (selectedVolunteer?: User) => void;
+};
+
+export const VolunteerDropdown = ({
+  volunteer,
+  onSelect,
+}: VolunteerDropdownProps) => {
+  const [loading, setLoading] = useState(false);
+  const [suggestionsList, setSuggestionsList] = useState<
+    TAutocompleteDropdownItem[] | null
+  >(null);
+  const searchRef = useRef<TextInput>(null);
+  const dropdownController = useRef<AutocompleteDropdownRef | undefined>();
+  const [selectedVolunteer, setSelectedVolunteer] = useState<User | undefined>(
+    volunteer
+  );
+
+  const insets = useSafeAreaInsets();
+  const theme = useTheme();
+  const styles = getStyles(theme, insets);
+
+  const encodeUserDetailsAsString = (user?: User) =>
+    `${user?.id};${user?.email};${user?.phone};${user?.username}`;
+  const decodeUserDetailsFromId = (userId: string) => userId.split(';');
+
+  const getSuggestions = useCallback(async (query: string) => {
+    setLoading(true);
+    const { success, users: volunteers } = await loadUsersRequest();
+    if (!success) alert('Failed to load users');
+
+    const tokenizedQuery = query.split(' ');
+    const queryRegexPattern = tokenizedQuery.reduce((acc, x) => acc + '|' + x);
+    const regex = new RegExp(queryRegexPattern, 'i');
+    const relevantVolunteers = volunteers.filter((x: User) =>
+      x.name.match(regex)
+    );
+
+    type UserIdNamePair = { id: string; title: string };
+
+    setSuggestionsList(
+      relevantVolunteers
+        .map((item: User) => ({
+          id: encodeUserDetailsAsString(item),
+          title: item.name,
+        }))
+        .sort((a: UserIdNamePair, b: UserIdNamePair) =>
+          a.title.localeCompare(b.title)
+        )
+    );
+    setLoading(false);
+  }, []);
+
+  const onClearPress = useCallback(() => {
+    setSuggestionsList(null);
+    setSelectedVolunteer(undefined);
+    onSelect(undefined);
+    dropdownController?.current?.open();
+  }, []);
+
+  const onBlur = () =>
+    dropdownController &&
+    dropdownController?.current?.setItem({
+      id: encodeUserDetailsAsString(selectedVolunteer),
+      title: selectedVolunteer?.name || null,
```

Comment:
why is the id an encoded version of the selected volunteer?

Reply:
That is because we need to display the details of a selected user within the dropdown component. Check lines 100-115.